lib/functions/logging/export-logs.sh

```diff
@@ -169,17 +169,39 @@ function export_ansi_logs() {
 	if [[ "${show_message_after_export:-"yes"}" != "skip" && "${ARMBIAN_INSIDE_DOCKERFILE_BUILD:-"no"}" != "yes" ]]; then
 		display_alert "ANSI log file built; inspect it by running:" "less -RS ${target_relative_to_src}"
 
-		# @TODO: compress...
-		declare paste_url="${PASTE_URL:-"https://paste.armbian.com/log"}"
+		# Define here the paste servers to use, in order: each will be tried in sequence until one works
+		declare -a paste_servers=("paste.armbian.com" "paste.armbian.de" "paste.next.armbian.com" "paste.armbian.eu")
+
+		# User can override by setting PASTE_SERVER_HOST=some.paste.server.com - it will be added as the first to try
+		if [[ "${PASTE_SERVER_HOST}" != "" ]]; then
+			display_alert "Using custom paste server" "${PASTE_SERVER_HOST}" "info"
+			paste_servers=("${PASTE_SERVER_HOST}" "${paste_servers[@]}")
+		fi
+
 		if [[ "${SHARE_LOG:-"no"}" == "yes" ]]; then
-			display_alert "SHARE_LOG=yes, uploading log" "uploading logs" "info"
-			declare logs_url="undetermined"
-			logs_url=$(curl --silent --data-binary "@${target_relative_to_src}" "${paste_url}" | xargs echo -n || true) # don't fail
-			display_alert "Log uploaded, share URL:" "${logs_url}" ""
-			# set output for GitHub Actions
-			github_actions_add_output logs_url "${logs_url}"
+			declare -i some_paste_server_worked=0
+			for paste_server in "${paste_servers[@]}"; do
+				declare paste_url="https://${paste_server}/log"
+				display_alert "SHARE_LOG=yes, uploading log" "uploading logs to '${paste_server}'" "info"
+				declare result_from_paste_server="undetermined"
+				result_from_paste_server=$(curl --silent --max-time 120 --data-binary "@${target_relative_to_src}" "${paste_url}" | xargs echo -n || true) # don't fail
```

Comment:
In _armbianmonitor_ there is also `--fail`. Not sure if necessary. https://github.com/armbian/build/blob/7184c5aa89c969323feac3903cd78cccb47a2ed4/packages/bsp/common/usr/bin/armbianmonitor#L182C1-L182C7

Reply:
No, not here; there, I use to exit status to determine if the thing failed or not.
Here I check if the response from curl (whatever it is) begins with the paste URL -- so no matter the exit code.
That is done due to the pain that is both getting the exit code _and_ capturing the output in bash.